lib/twilio/audiohelper.ts

```diff
@@ -275,6 +285,19 @@ class AudioHelper extends EventEmitter {
     if (isEnumerationSupported) {
       this._initializeEnumeration();
     }
+
+    // NOTE (kchoy): Currently microphone permissions are not supported in firefox.
+    // https://github.com/mozilla/standards-positions/issues/19#issuecomment-370158947
+    navigator.permissions.query({ name: 'microphone' }).then((microphonePermissionStatus) => {
+      if (microphonePermissionStatus.state !== 'granted') {
+        const handleStateChange = () => {
+          this._updateAvailableDevices();
```

Comment:
I see that you removed the line where we unsubscribe to the change event. Should we call `this._stopMicrophonePermissionListener();` here instead?

Reply:
oops yes will put it here as well